lib/visenze_productsearch.dart

```diff
@@ -0,0 +1,87 @@
+library visenze_productsearch_sdk;
+
+import 'package:http/http.dart';
+import 'package:visenze_productsearch_sdk/src/productsearch_client.dart';
+import 'package:visenze_tracking_sdk/visenze_tracker.dart';
+
+class VisenzeProductSearch {
+  final String _appKey;
+  final String _placementId;
+  late final ProductSearchClient _client;
+  late final VisenzeTracker _tracker;
+
+  /// Factory for creating [VisenzeProductSearch]
+  ///
+  /// Require authentication params [appKey] and [placementId]
+  ///
+  /// If [uid] is provided, set tracker user id to [uid]
+  ///
+  /// If [timeout] is provided, request timeout is set to [timeout] in ms
+  static Future<VisenzeProductSearch> create(String appKey, String placementId,
+      {String? uid, bool? useStaging, int? timeout}) async {
+    var ps = VisenzeProductSearch._create(appKey, placementId);
+    await ps._init(uid: uid, useStaging: useStaging, timeout: timeout);
+    return ps;
+  }
+
+  /// Get the current session id
+  String get sessionId {
+    return _tracker.sessionId;
+  }
+
+  /// Get the current user id
+  String get userId {
+    return _tracker.userId;
+  }
+
+  /// Set the current user id to the provided [uid]
+  set userId(String uid) {
+    _tracker.userId = uid;
+  }
+
+  /// Reset the current session and return the new session id
+  String resetSession() {
+    return _tracker.resetSession();
+  }
+
+  String? get lastSuccessQueryId {
+    return _client.lastSuccessQueryId;
+  }
+
+  /// Send a request to ViSenze analytics server with event name [action] and provided [queryParams]
+  Future<void> sendEvent(
+      String action, Map<String, dynamic> queryParams) async {
+    final params = {...queryParams}..addAll(_client.getCommonParams());
```

Comment:
just to confirm if the `client.getCommonParams()` will overwrite `queryParams`?

for example: `queryParams.va_sdk = custom_sdk` & `commonParams.va_sdk = flutter_sdk_v1`
the end result is `va_sdk=flutter_sdk_v1` right?

Reply:
is it possible to add a test case for this scenario?

Reply:
yes, it will overwrite. 
Is it better that the custom queryParams to take priority? Since the common only contains version number.

Reply:
no, the common vars should be taking priority. so it is ok as it is now.
this is assuming that the common vars are information we want to track and not intended for customer to change like device info/sdk version/../etc